src/components/advanced-search/advanced-search-retrieve.vue

```diff
@@ -0,0 +1,170 @@
+
+    <!-- Phone input -->
+    <v-row class="mt-6" no-gutters>
+      <v-col>
+        <v-text-field
+          filled persistent-hint
+          id="phone-number-text-field"
+          class="copy-normal"
+          label="Applicant's Phone Number"
+          hint="Example: 555-555-5555"
+          v-model="phoneNumber"
+          :rules="phoneRules"
+          @input="handleExistingRequestSearch('phoneNumber', $event)"
+        />
+      </v-col>
+    </v-row>
+
+    <v-row>
+      <v-col class="copy-normal font-weight-bold ml-4 py-0"> or </v-col>
+    </v-row>
+
+    <!-- Email input -->
+    <v-row>
+      <v-col>
+        <v-text-field
+          filled persistent-hint
+          id="email-address-text-field"
+          class="copy-normal"
+          label="Applicant's Notification Email"
+          hint="Example: name@email.com"
+          v-model="emailAddress"
+          :rules="emailRules"
+          @input="handleExistingRequestSearch('emailAddress', $event)"
+        />
+      </v-col>
+    </v-row>
+
+  </v-form>
+</template>
+
+<script lang='ts'>
+import { Component, Emit, Prop, Vue, Watch } from 'vue-property-decorator'
+import AdvancedSearchDates from '@/components/advanced-search/advanced-search-dates.vue'
+
+// Interfaces
+import { FormType, NameRequestI } from '@/interfaces'
+import { Action, Getter } from 'vuex-class'
+import { ActionBindingIF } from '@/interfaces/store-interfaces'
+import NamexServices from '@/services/namex.services'
+
+@Component({
+  components: {
+    AdvancedSearchDates
+  }
+})
+export default class AdvancedSearchRetrieve extends Vue {
+  // Refs
+  $refs!: {
+    advancedSearchRetrieveForm: FormType
+  }
+
+  /** Prompt the form to pass up the form data. */
+  @Prop() readonly promptSubmit: boolean
+
+  // Global getter
+  @Getter getNr!: Partial<NameRequestI>
+
+  // Global setters
+  @Action findNameRequest!: ActionBindingIF
+  @Action setExistingRequestSearch!: ActionBindingIF
+  @Action setNameRequest!: ActionBindingIF
+  @Action setDisplayedComponent!: ActionBindingIF
+
+  // Local properties
+  private isValid = false
+  private phoneNumber = ''
+  private emailAddress = ''
+  private errorMessage = ''
+
+  /** Validation rules for Applicant phone and email. */
+  private phoneRules = []
+  private emailRules = []
+
+  /** Apply the field rules to request search inputs. */
+  private applyRules (): void {
+    this.phoneRules = (this.phoneNumber || !this.emailAddress)
+      ? [
+        v => !!v || 'Please enter a phone number',
+        v => (v?.length <= 30) || 'Cannot exceed 30 characters'
+      ]
+      : []
+
+    this.emailRules = (!this.phoneNumber || this.emailAddress)
+      ? [
+        v => !!v || 'Please enter an email',
+        v => (/.+@.+\..+/.test(v)) || 'A valid email is required'
+      ]
+      : []
+  }
+
+  /** Clear field level validations. */
+  private clearValidations (): void {
+    this.$refs.advancedSearchRetrieveForm.resetValidation()
```

Comment:
You may have to await this.

Reply:
Will have a look but i didn't notice any issues here!

Reply:
I believe there's no visible issue because the subsequent code doesn't depend on it, but I highly suspect resetting validation takes a few cycles -- like `validate()`.

I quickly searched the code base and there's actually no indication that `validate` is async, although we know from experience that we need to await the next tick to get the result we want... more in the comment below.

I can't find `resetValidation()` anywhere in the code base. (Are we sure it works? lol) On the assumption that it works like validation, I think it be would safest to add an "await nextTick" after the resetValidation call...

I see this app calls `resetValidation` in a few other places without waiting for it to finish -- again, no visible issue because of the current code flow. So I dunno: stay consistent and risk (another) async issue in this app, or go with a gut feel? What do you think?

Reply:
I prefer to await as well. I think that would be the prudent approach for defensive coding :) 
Updated!